### PR TITLE
Json matcher

### LINKED ISSUE #559: json_params_matcher does not support general JSONs (e.g. lists of dicts)
### Describe the bug

The implementation of `json_params_matcher` does not support arbitrarily nested JSON objects, such as lists of dicts. 

It says so in the signature: 

`def json_params_matcher(params: Optional[Dict[str, Any]]) -> Callable[..., Any]:`

and the simple check of `params_dict == json_body` is not enough. Additionally, an Exception is raised in this section, as a dict is expected.

`
            if not valid:
                reason = "request.body doesn't match: {} doesn't match {}".format(
                    _create_key_val_str(json_body), _create_key_val_str(params_dict)
                )
`

### Additional context

I can provide a PR, but there are different ways to tackle this:

- Custom function to recursively check => no dependencies
- Adopting something like https://github.com/seperman/deepdiff or https://github.com/xlwings/jsondiff to support with the extraction of reasons for JSONs being not equal

As this might entail additional dependencies, I'd like to get your opinion on this before submitting a PR.

### Version of `responses`

0.21.0

### Steps to Reproduce

```python
import json
from unittest.mock import Mock

from responses.matchers import _create_key_val_str, json_params_matcher


json_a = [{"a": "b"}]
json_b = [{"a": "b", "c": "d"}]
mock_request = Mock(body=json.dumps(json_b))
result = json_params_matcher(json_a)(mock_request)
print(result)

```

### Expected Result

`result` should be (False, "<reason>"), no error is thrown

### Actual Result

AttributeError is raised: 'list' object has no attribute 'keys'

### PR DESCRIPTION
**Pull request by beliaev-maksim**
closes #559 

I do not think we need to go for over-complicated comparisons of objects. User can extend functionality by applying custom matcher.
However, I do agree that lists should be supported as valid JSON

### CODE REVIEW

**Comment by beliaev-maksim** [comment]
hi @chadwhitacre 

I think something is wrong with FOSSA pipeline. 22min for check is too much. I have already seen similar for other PRs. Can you please check?

File: responses/matchers.py [hunk 1]
```diff
@@ -121,22 +122,30 @@ def json_params_matcher(
     def match(request: PreparedRequest) -> Tuple[bool, str]:
         reason = ""
         request_body = request.body
-        params_dict = params or {}
+        json_params = (params or {}) if not isinstance(params, list) else (params or [])
```
Comment (markstory) [inline responses/matchers.py:125]:
```suggestion
        json_params = (params or {}) if not isinstance(params, list) else params
```

Do we need the second `or`? If params is a list that is falsey, it is already an empty list.

Comment (beliaev-maksim) [reply responses/matchers.py:125]: you are right, thanks

**Comment by chadwhitacre** [comment]
Glad this landed. Sorry for the FOSSA noise, we ended up [turning it off](https://github.com/getsentry/team-ospo/issues/4#issuecomment-1215091008) for now.

**Pull request merged**